tools/config/__init__.py

```diff
@@ -710,8 +710,19 @@ def _generate_bootloader_build(self, rom_start, rom_size):
             if part.minaddr() != rom_start:
                 raise ConfigException("bootloader executable does not "
                                       "start at 0x%x" % rom_start)
-            part_size = (part.maxaddr() - part.minaddr()) + 1
-            part_size = Config._align_ceiling(rom_start + part_size, self.sectors) - rom_start
+
+            # find the last valid address that's within rom_end and use that
+            # to compute the bootloader size
+            end_address = None
+            for each in part.segments():
+                if (each[1] < rom_end):
+                    end_address = each[1]
+                else:
+                    break
+            if end_address == None:
+                raise ConfigException("bootloader segments don't fit within rom region")
+            part_size = Config._align_ceiling(rom_start + (end_address - start), self.sectors) - rom_start
```

Comment:
`start` == `rom_start`

```suggestion
            part_size = Config._align_ceiling(end_address, self.sectors) - rom_start
```